lib/oli_web/live/workspaces/course_author/bibliography_live.ex

```diff
@@ -1,30 +1,64 @@
 defmodule OliWeb.Workspaces.CourseAuthor.BibliographyLive do
   use OliWeb, :live_view
 
+  alias Oli.Accounts
+  alias OliWeb.Common.React
+
+  on_mount {OliWeb.LiveSessionPlugs.AuthorizeProject, :default}
```

Comment:
This on_mount seems general enough that it could maybe live in the router as part of the live_session?

Reply:
Good observation @eliknebel. That `on_mount` is only necessary when working with a project, but the instructor is also in the scope of the workspace's **live_session**, as shown in the snippet below. I'm working on a fix for that in PR https://github.com/Simon-Initiative/oli-torus/pull/5069 where we will have 2 live_sessions: one for `course_author` and another one for the `instructor`. Would you like me to include it here, or should I wait until the fix is complete?

https://github.com/Simon-Initiative/oli-torus/blob/85ad23911ef9f8c31b3a1aec043f6d04f60e9305/lib/oli_web/router.ex#L789-L827

Reply:
We can handle it in the follow-up PR